pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go

```diff
@@ -93,15 +110,35 @@ func (c *BootstrapTeardownController) removeBootstrap() error {
 		return nil
 	}
 
-	etcdReady, err := isEtcdBootstrapped(currentEtcdOperator)
+	// check if bootstrap is already removed
+	if v1helpers.IsOperatorConditionTrue(currentEtcdOperator.Status.Conditions, ConditionBootstrapRemoved) {
```

Comment:
No.  This controller should work based on reality, not based on some previously reported status.

Reply:
I can replace that with `!c.isMember("etcd-bootstrap")`

But then again, if `etcd-bootstrap` is a member or not is reported another controller [1]. So not sure if the change addresses pitfalls you had in mind. Can I ask what you think can go wrong with the current approach of looking up the previously reported condition? @deads2k 

1. https://github.com/openshift/cluster-etcd-operator/blob/be85b9fb56b9153500ff64e7b6cb0adbc447b1c5/pkg/operator/configobservation/etcd/observe_etcd.go#L385:24